File: LeetCode/Top Interview Questions/Array/Easy/Single_Number_hari.cpp
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        sort(nums.begin(),nums.end());
        int i=0;
        while(i<(nums.size()-1))
        {
           if(nums[i]!=nums[i+1])
           {
               return nums[i];
           }
            i+=2;
        }
        return nums[i];
        
    }
};

// second attempt
class Solution {
public:
    int singleNumber(vector<int>& nums) {
        sort(nums.begin(), nums.end());
        int len = nums.size();
        int i,j;
        i=0;
        j=1;
        while(j<len)
        {
            if(nums[i] != nums[j])
                return nums[i];
            i+=2;
            j+=2;
        }
        return nums[i];
    }
};

File: LeetCode/Top Interview Questions/Array/Easy/Single_Number_hari.py
class Solution(object):
    def singleNumber(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        a = 0
        for i in nums:
            a ^= i
        return a


class Solution:
    def singleNumber(self, nums: List[int]) -> int:
        hash ={}
        for i in nums:
            if i in hash:
                hash[i]+=1
            else:
                hash[i]=1
        for i in hash:
            if hash.get(i) == 1:
                return i
        return 0
